### ISSUE TITLE
`prefer-set-size` rule does not catch `Array.from()` usage for converting a set to an array

### ISSUE THREAD

**Issue by connorshea** [sindresorhus/eslint-plugin-unicorn#2843]
`Array.from()` on a `Set` doesn't get noticed by the prefer-set-size rule as a conversion from a set to an array

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/rules/prefer-set-size.js

```js
// This passes when it shouldn't:
Array.from(new Set(array)).length
```

**Comment by meetpotdar777** [comment]
I see there is already a PR in progress for this, but I wanted to share some thoughts on the logic for the pseudo-code to ensure we catch all edge cases...

> "To catch this edge case, we should look for a `MemberExpression` where the property is `length` and the object is a `CallExpression` to `Array.from()`. If the argument inside `Array.from()` is a `NewExpression` of a `Set`, we should trigger the lint error.
> Example to catch:
> ```js
> // Suggest changing this:
> const count = Array.from(new Set(items)).length;
> // To this:
> const count = new Set(items).size;
> ```"
> 
> ```
> 
> 
---
